pkg/sql/create_table.go

```diff
@@ -2426,27 +2425,26 @@ func newTableDesc(
 }
 
 // newRowLevelTTLScheduledJob returns a *jobs.ScheduledJob for row level TTL
-// for a given table.
+// for a given table. newRowLevelTTLScheduledJob assumes that
+// tblDesc.RowLevelTTL is not nil.
 func newRowLevelTTLScheduledJob(
-	env scheduledjobs.JobSchedulerEnv,
-	owner username.SQLUsername,
-	tblID descpb.ID,
-	ttl *catpb.RowLevelTTL,
+	env scheduledjobs.JobSchedulerEnv, owner username.SQLUsername, tblDesc *tabledesc.Mutable,
 ) (*jobs.ScheduledJob, error) {
 	sj := jobs.NewScheduledJob(env)
-	sj.SetScheduleLabel(fmt.Sprintf("row-level-ttl-%d", tblID))
+	sj.SetScheduleLabel(fmt.Sprintf("row-level-ttl-%s", tblDesc.GetName()))
 	sj.SetOwner(owner)
 	sj.SetScheduleDetails(jobspb.ScheduleDetails{
 		Wait: jobspb.ScheduleDetails_WAIT,
 		// If a job fails, try again at the allocated cron time.
 		OnError: jobspb.ScheduleDetails_RETRY_SCHED,
 	})
 
-	if err := sj.SetSchedule(ttl.DeletionCronOrDefault()); err != nil {
+	// TODO: Is it better to check that RowLevelTTL is not nil and return an error?
```

Comment:
nit: `TODO(username): `

Reply:
+1 to dt's point

Also, If we're setting schedule for a nil TTL, it's a critical bug that we should let it fail hard. but it's not nice to panic. We could return a assertion error with `errors.AssertionFailedf` to feel safe.

Reply:
Sorry for the confusion. This was [intended to be a discussion at review time](https://github.com/cockroachdb/cockroach/pull/107294#issuecomment-1644436745). I'll use `TODO(BEFORE MERGING)` in the future!

> Also, If we're setting schedule for a nil TTL, it's a critical bug that we should let it fail hard. but it's not nice to panic. We could return a assertion error with errors.AssertionFailedf to feel safe.

Agreed that panicking isn't great. More generally, is the style in CRDB is to assume that programmers are doing the right thing or is it to be paranoid? Because of how this function got refactored and how go's type system works, callers could accidentally cause nil dereferences. I find myself running into these types of cases frequently. In general I prefer to write types in such a way that this can't happen but that's not always on option (See: this PR). My personal rule is to be paranoid at boundaries and expect well shaped data inside of said boundaries. I don't really have a sense of what is or is not considered to be a boundary yet. `sql.CreateRowLevelTTLScheduledJob` passes `tblDesc` right through, so we could do a check at the package level but this metaphorical boundary might exist somewhere else instead. What would either of you do for this specific situation and does that follow a more generalizable rule?

Reply:
mhm, I think you already made good points here, we can do the check in `CreateRowLevelTTLScheduledJob` given it's a public function at the package level. I personally often makes assumptions in non-public functions, of course sometimes do such kind of validations in a helper.

Reply:
Thanks! Added the check to `CreateRowLevelTTLScheduleJob`.